LanguageFeatures/Patterns/type_inference_A25_t01.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// To type check a pattern p being matched against a value of type M:
+/// ...
+/// Variable:
+///
+/// If the variable has a type annotation, the required type of p is that type,
+/// as is the static type of the variable introduced by p.
+///
+/// Else the required type of p is M, as is the static type of the variable
+/// introduced by p
+///
+/// @description Check that if a variable pattern `p` has a type annotation,
+/// then the required type of `p` is that type. Test that there is no match in
+/// a refutable context if the matched value type is not assignable to the type
+/// of the variable pattern
```

Comment:
OK. This is a property of the dynamic semantics ("there is a match / there is no match" is a question that we can only answer at run time). That seems to be out of topic here, cf. the title 'Type checking and pattern required type tests'. However, the situation where the matched value type is not assignable to the required type of the variable pattern _is_ relevant to compile time: That's not an error in a refutable context, but it is an error in an irrefutable context.

Perhaps it's just a typo in line 17: 'Test that there is no compile-time error in ...'?